client/buffered_client.go

```diff
@@ -0,0 +1,70 @@
+package client
+
+import "time"
+
+type BufferConfig struct {
+	FlushMaxPoints   int
+	FlushMaxWaitTime time.Duration
+}
+
+func NewBufferedClient(clientConfig Config, bufferConfig BufferConfig) (bufferedClient *BufferedClient, err error) {
+	client, err := NewClient(clientConfig)
+	if err != nil {
+		return
+	}
+	bufferedClient = &BufferedClient{
+		Client:       client,
+		bufferConfig: bufferConfig,
+		ingestChan:   make(chan Point, bufferConfig.FlushMaxPoints/3),
+		flushTimer:   time.NewTimer(bufferConfig.FlushMaxWaitTime),
+		pointsBuf:    make([]Point, bufferConfig.FlushMaxPoints),
+		pointsIndex:  0,
+	}
+	go bufferedClient.ingestAndFlushLoop()
```

Comment:
You should just pass a channel to `ingestAndFlushLoop()` called, say, `done`, and then close than channel when the system is shutting down. You'll need to add `done` to your `select` block. You also need to use a wait group to sync the whole thing i.e. the standard pattern when coding stuff like this in Go.